engine/src/cmd/planet.cpp

```diff
@@ -709,7 +709,7 @@ void Planet::Draw(const Transformation &quat, const Matrix &m) {
     }
     GFXLoadIdentity(MODEL);
     for (unsigned int i = 0; i < lights.size(); i++) {
-        GFXSetLight(lights[i], POSITION, GFXColor(cumulative_transformation.position.Cast()));
```

Comment:
I don't think there is anything to be done here. I haven't delved into `cumulative_transformation.position.Cast()` but they're clearly using an actual vector here, not as a convenience. I would simply roll with it and do something like:

```
Vector v = cumulative_transformation.position.Cast();
GFXColor c = GFXColor(v.i,v.j,v.k);
...
```

Reply:
They are setting the POSITION property of the light... with a GFXColor... constructed from a Vector. This is doubly wrong LOL. We should just add another overload of the GFXSetLight method to set properties of lights -- an overload that takes Vector instead of GFXColor. Which is what I have done.

Reply:
Another overload is needed for GFXLight::SetProperties as well, which GFXSetLight calls under the hood.

Reply:
If it's a color-related property, set it using a GFXColor; if it's coordinates-related or similar, set it using a Vector. At least, that's how I think it should generally be done.

Reply:
Yeah. I assumed it was a color they were setting there. Silly me.
I agree.